WIP: initial commit of route_func and common library

rasibley opened the pull request:
This PR includes the networking common library and route_func test case. Currently we use RhtsRequires to pull in the library in the Makefile, but instead we need to port the library and then fetch it in the metadata file using 'Requires=networking/common'. I'm still waiting on review from the test maintainers, however would like initial feedback while we wait. @veruu @idorax @ftjahjad

ftjahjad commented:
@rasibley 
I think is very clean to have  "networking common" library. :100: 

seeing another "networking tunnel common library" for tunnel  basic test that I need to port .

veruu commented on networking/common/lib/protocol/netns.sh:
```diff
@@ -0,0 +1,254 @@
+	# Setup Server
+	vcommon server "rm -f /tmp/test_*; $VTOPO; get_test_iface"
+	SERVER_IFACE=$(tail -n1 /tmp/test_iface | tr -d '\r\n')
+	[ ! "$SERVER_IFACE" ] && test_fail "No SERVER IFACE" && return 1
```
Is this really a failure, or is it just blocking the test from running?

veruu commented on networking/common/lib/protocol/netns.sh:
```diff
@@ -0,0 +1,254 @@
+	# Setup Client
+	vcommon client "rm -f /tmp/test_*; $VTOPO; get_test_iface"
+	CLIENT_IFACE=$(tail -n1 /tmp/test_iface | tr -d '\r\n')
+	[ ! "$CLIENT_IFACE" ] && test_fail "No CLIENT IFACE" && return 1
```
Same here

veruu commented on networking/common/obsolete.sh:
```diff
@@ -0,0 +1,80 @@
+		if [ "$3" == "check" ];then
+			if [ $LASTSTATE -ne 0 ];then
+				echo -e "Command # $1 failed\n" | tee -a "$OUTPUTFILE"
+				test_warn "Command_failed"
```
Should this be a failure instead? Or can it mean the preparation for the test could have failed too (in which case it should abort)?

veruu commented on networking/common/include.sh:
```diff
@@ -0,0 +1,375 @@
+		sleep 5
+		let timeout=timeout-5
+	done
+	if [ $timeout -le 0 ]; then test_warn "rhts-sync-set $HOSTNAME $message failed"; fi
```
Should this also abort?

veruu commented on networking/route/route_func/common/include.sh:
```diff
@@ -0,0 +1,391 @@
+		sleep 5
+		let timeout=timeout-5
+	done
+	if [ $timeout -le 0 ]; then test_warn "rhts-sync-set $HOSTNAME $message failed"; fi
```
Same here

rasibley commented:
Thanks @veruu  ! I have modified test_warn() function in the include.sh file to abort so it should cover all cases where where this is used (infra failures), also removed some files which shouldn't have been there.

veruu commented:
As the points were resolved, let's merge this and we can do any changes later if needed.
